src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/InterfaceMemberImpliedModifierCheck.java

```diff
@@ -0,0 +1,423 @@
+    }
+
+    /**
+     * Setter to control whether to enforce that {@code public} is explicitly coded
+     * on interface methods.
+     * @param violateImpliedPublicMethod
+     *        True to perform the check, false to turn the check off.
+     */
+    public void setViolateImpliedPublicMethod(boolean violateImpliedPublicMethod) {
+        this.violateImpliedPublicMethod = violateImpliedPublicMethod;
+    }
+
+    /**
+     * Setter to control whether to enforce that {@code abstract} is explicitly coded
+     * on interface methods.
+     * @param violateImpliedAbstractMethod
+     *        True to perform the check, false to turn the check off.
+     */
+    public void setViolateImpliedAbstractMethod(boolean violateImpliedAbstractMethod) {
+        this.violateImpliedAbstractMethod = violateImpliedAbstractMethod;
+    }
+
+    /**
+     * Setter to control whether to enforce that {@code public} is explicitly coded
+     * on interface nested types.
+     * @param violateImpliedPublicNested
+     *        True to perform the check, false to turn the check off.
+     */
+    public void setViolateImpliedPublicNested(boolean violateImpliedPublicNested) {
+        this.violateImpliedPublicNested = violateImpliedPublicNested;
+    }
+
+    /**
+     * Setter to control whether to enforce that {@code static} is explicitly coded
+     * on interface nested types.
+     * @param violateImpliedStaticNested
+     *        True to perform the check, false to turn the check off.
+     */
+    public void setViolateImpliedStaticNested(boolean violateImpliedStaticNested) {
+        this.violateImpliedStaticNested = violateImpliedStaticNested;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.METHOD_DEF,
+            TokenTypes.VARIABLE_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.CLASS_DEF,
+            TokenTypes.ENUM_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ScopeUtil.isInInterfaceBlock(ast) && !ScopeUtil.isInCodeBlock(ast)) {
+            switch (ast.getType()) {
+                case TokenTypes.METHOD_DEF:
+                    processMethod(ast);
+                    break;
+                case TokenTypes.VARIABLE_DEF:
+                    processField(ast);
+                    break;
+                case TokenTypes.CLASS_DEF:
+                case TokenTypes.INTERFACE_DEF:
+                case TokenTypes.ENUM_DEF:
+                    processNestedType(ast);
+                    break;
+                default:
+                    throw new IllegalStateException(ast.toString());
+            }
+        }
+    }
+
+    /**
+     * Check method on interface.
```

Comment:
`Check method on interface` => `Checks method in interface`

Reply:
Done